jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticUtils.java

```diff
@@ -120,6 +143,52 @@ private static DiagnosticProperties getHeapStatistics(String diagnosticCommand)
 		}
 		return DiagnosticProperties.makeStringResult(result);
 	}
+	
+	private static DiagnosticProperties getThreadInfo(String diagnosticCommand) {
+		String newline = System.lineSeparator();
+		DiagnosticProperties result = null;
+		boolean okay = true;
+		boolean addSynchronizers = false;
+		String[] parts = diagnosticCommand.split(DIAGNOSTICS_OPTION_SEPARATOR);
+		if (parts.length > 2) {
+			okay = false;
+		} else if (parts.length == 2) {
+			String option = parts[1];
+			/* there is only one option. */
+			if (option.startsWith(THREAD_LOCKED_SYNCHRONIZERS_OPTION)) {
+				if ((THREAD_LOCKED_SYNCHRONIZERS_OPTION.length() == option.length()) /* exact match */
+						|| option.toLowerCase().endsWith("=true")) { //$NON-NLS-1$
```

Comment:
What does this support?

Reply:
Comment updated. Let me know if it isn't clear.

Reply:
Let me know when the commit is pushed.

Reply:
Please answer my question.